### PR TITLE
Add more scenarios for network undefine

### PR DESCRIPTION
**Pull request by yalzhang**
Add more scenarios about undefine networks to cover different status of
networks.

Signed-off-by: yalzhang <yalzhang@redhat.com>

### CODE REVIEW

**Comment by yalzhang** [comment]
run all cases manually successfully.

JOB ID     : 76f24f324de569de49755d3249b6b6e32e9be77a
JOB LOG    : /root/avocado/job-results/job-2018-05-10T21.16-76f24f3/job.log
 (01/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.normal_test.new_network.non_acl: PASS (48.52 s)
 (02/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.normal_test.new_network.acl_test: PASS (61.45 s)
 (03/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.normal_test.new_network.state_persistent_active: PASS (55.40 s)
 (04/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.normal_test.new_network.state_transient_active: PASS (52.77 s)
 (05/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.normal_test.new_network.state_persistent_inactive: PASS (50.49 s)
 (06/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.normal_test.default_network.non_acl: PASS (56.12 s)
 (07/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.normal_test.default_network.acl_test: PASS (58.97 s)
 (08/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_network.invalid_uuid: PASS (44.24 s)
 (09/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_network.invalid_netname: PASS (47.20 s)
 (10/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.acl_test.define_acl: PASS (57.66 s)
 (11/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.acl_test.undefine_acl: PASS (58.27 s)
 (12/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_options.net_define.none_option: PASS (45.06 s)
 (13/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_options.net_define.not_exist_option: PASS (45.67 s)
 (14/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_options.net_define.additional_arg: PASS (45.63 s)
 (15/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_options.net_define.additional_option: PASS (45.77 s)
 (16/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_options.net_undefine.none_option: PASS (46.42 s)
 (17/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_options.net_undefine.not_exist_option: PASS (44.28 s)
 (18/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_options.net_undefine.additional_arg: PASS (45.44 s)
 (19/19) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_options.net_undefine.additional_option: PASS (46.34 s)

File: libvirt/tests/cfg/virsh_cmd/network/virsh_net_define_undefine.cfg [hunk 1]
```diff
@@ -25,8 +25,18 @@
                             action_lookup = "connect_driver:QEMU network_name:foobar"
                             unprivileged_user = "EXAMPLE"
                             virsh_uri = "qemu:///system"
-                        - state_test:
+                        - state_persistent_active:
                             check_states = "yes"
+                            net_persistent = "yes"
+                            net_active = "yes"
+                        - state_transient_active:
+                            check_states = "yes"
```
Comment (chunfuwen) [inline libvirt/tests/cfg/virsh_cmd/network/virsh_net_define_undefine.cfg]: check_states = "yes" is the same across variant ,and can be shared

Comment (yalzhang) [reply libvirt/tests/cfg/virsh_cmd/network/virsh_net_define_undefine.cfg]: no, there are 5 variants here, only 3 of them use the ' check_states = "yes" '

File: libvirt/tests/src/virsh_cmd/network/virsh_net_define_undefine.py [hunk 2]
```diff
@@ -183,17 +185,41 @@ def run(test, params, env):
...
                 result_info.append("Found wrong network state after restarting"
                                    " libvirtd: %s" % str(net_state))
-            # Undefine an active network and check state
+            logging.debug("undefine network:")
+            # prepare the network status
+            if not net_persistent:
+                virsh.net_undefine(net_name, ignore_status=False)
+            if not net_active:
+                virsh.net_destroy(net_name, ignore_status=False)
             undefine_status = virsh.net_undefine(undefine_options, undefine_extra,
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/network/virsh_net_define_undefine.py]: line 190 also do net_undefine if not net_persistent, I assume there is duplicate code here

File: libvirt/tests/src/virsh_cmd/network/virsh_net_define_undefine.py [hunk 3]
```diff
@@ -183,17 +185,41 @@ def run(test, params, env):
...
                                                  **virsh_dargs).exit_status
+            # when undefine cmd fail
+            if undefine_status:
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/network/virsh_net_define_undefine.py]: line 197 to 202 can be implemented by libvirt.check_exit_status(ret, expected_error)

File: libvirt/tests/src/virsh_cmd/network/virsh_net_define_undefine.py (hunk 2 above)
Comment (yalzhang) [reply libvirt/tests/src/virsh_cmd/network/virsh_net_define_undefine.py]: no, line 191 is to prepare the env, line 194 it to execute the test cmd.
It is a negative case to undefine transient network in line 194.
If the network is persistent, in line 191, we need to undefine it to make it to be transisent

File: libvirt/tests/src/virsh_cmd/network/virsh_net_define_undefine.py [hunk 4]
```diff
@@ -183,17 +185,41 @@ def run(test, params, env):
...
+            # when cmd succeed
             if not undefine_status:
                 net_state = virsh_instance.net_state_dict()
-                if (not net_state[net_name]['active'] or
-                        net_state[net_name]['autostart'] or
-                        net_state[net_name]['persistent']):
-                    fail_flag = 1
-                    result_info.append("Found wrong network states for "
-                                       "undefined netowrk: %s" % str(net_state))
+                if net_persistent and net_active:
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/network/virsh_net_define_undefine.py]: Suggest reconstruct if statement, below is one reference sample:
```
if net_persistent:
      net_state_active = not net_state[net_name]['active'] or net_state[net_name]['autostart']
      if net_active and net_state_active:
          fail_flag = 1
else:
       fail_flag = 1
       xx
if not net_active and net_name in net_state:
      fail_flag = 1
```

**Pull request merged**